src/data/fin.lean

```diff
@@ -827,6 +827,48 @@ lemma exists_fin_succ {P : fin (n+1) → Prop} :
 ⟨λ ⟨i, h⟩, fin.cases or.inl (λ i hi, or.inr ⟨i, hi⟩) i h,
   λ h, or.elim h (λ h, ⟨0, h⟩) $ λ⟨i, hi⟩, ⟨i.succ, hi⟩⟩
 
+/-- Define `f : Π i : fin (m + n), C i` by separately handling the cases `i = cast_add n i`, 
+`j : fin m` and `i = cast_add_right m j`, `j : fin n`. -/
+@[elab_as_eliminator, elab_strategy]
```

Comment:
Note there is [a zulip thread ](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/elab_strategy/near/319192755) about this use of `elab_strategy`.